api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -253,7 +299,13 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     if (user == null) {
       workspaces = new ArrayList<>();
     } else {
-      workspaces = workspaceService.dao.findByCreatorOrderByNameAsc(userProvider.get());
+      Set<org.pmiops.workbench.db.model.WorkspaceUserRole> usersWorkspaces;
+      usersWorkspaces = user.getWorkspaceUserRoles();
+      Iterator<org.pmiops.workbench.db.model.WorkspaceUserRole> workspacesIterator = usersWorkspaces.iterator();
```

Comment:
Nit: shorter would be:

for (WorkspaceUseRole userRole : user.getWorkspaceUserRoles()) {
  workspaces.add(userRole.getWorkspace());
}